packages/mui-material/src/ListItemText/ListItemText.d.ts

```diff
@@ -3,11 +3,36 @@ import { SxProps } from '@mui/system';
 import { InternalStandardProps as StandardProps, Theme } from '..';
 import { TypographyProps } from '../Typography';
 import { ListItemTextClasses } from './listItemTextClasses';
+import { CreateSlotsAndSlotProps, SlotProps } from '../utils/types';
+
+export interface ListItemTextSlots {
+  /**
+   * The component that renders the main content.
+   * @default span
+   */
+  primary?: React.ElementType;
+  /**
+   * The component that renders the main content.
+   * @default p
+   */
+  secondary?: React.ElementType;
+}
+
+export type ListItemTextSlotsAndSlotProps = CreateSlotsAndSlotProps<
+  ListItemTextSlots,
+  {
+    primary: SlotProps<React.ElementType<HTMLSpanElement>, {}, ListItemTextOwnerState>;
+    secondary: SlotProps<React.ElementType<HTMLParagraphElement>, {}, ListItemTextOwnerState>;
+  }
```

Comment:
Please check https://github.com/mui/material-ui/pull/44570/files#diff-655ffb1d0cc5de911e82c2f708a9a1617a8c9acf40f7832ee468a98353bbdf4eR83

I think it's nice to add description to the `slotProps` with links to the API page in case it's a another component. The DX is a lot better.